packages/mds-utils/state-machine.ts

```diff
@@ -1,58 +1,136 @@
-    [VEHICLE_EVENTS.deregister]: VEHICLE_STATES.inactive
+    [VEHICLE_EVENTS.agency_drop_off]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.decommissioned]: [VEHICLE_STATES.removed],
+    [VEHICLE_EVENTS.provider_drop_off]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.unspecified]: [VEHICLE_STATES.available]
   },
   [VEHICLE_STATES.reserved]: {
-    [VEHICLE_EVENTS.trip_start]: VEHICLE_STATES.trip,
-    [VEHICLE_EVENTS.cancel_reservation]: VEHICLE_STATES.available
-  },
-  [VEHICLE_STATES.trip]: {
-    [VEHICLE_EVENTS.trip_leave]: VEHICLE_STATES.elsewhere,
-    [VEHICLE_EVENTS.trip_end]: VEHICLE_STATES.available
+    [VEHICLE_EVENTS.comms_lost]: [VEHICLE_STATES.unknown],
+    [VEHICLE_EVENTS.missing]: [VEHICLE_STATES.unknown],
+    [VEHICLE_EVENTS.reservation_cancel]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.trip_start]: [VEHICLE_STATES.on_trip],
+    [VEHICLE_EVENTS.unspecified]: [VEHICLE_STATES.available]
   },
-  [VEHICLE_STATES.unavailable]: {
-    [VEHICLE_EVENTS.service_start]: VEHICLE_STATES.available,
-    [VEHICLE_EVENTS.deregister]: VEHICLE_STATES.inactive,
-    [VEHICLE_EVENTS.agency_pick_up]: VEHICLE_STATES.removed,
-    [VEHICLE_EVENTS.provider_pick_up]: VEHICLE_STATES.removed
+  [VEHICLE_STATES.unknown]: {
+    [VEHICLE_EVENTS.agency_drop_off]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.agency_pick_up]: [VEHICLE_STATES.removed],
+    [VEHICLE_EVENTS.comms_restored]: [
+      VEHICLE_STATES.available,
+      VEHICLE_STATES.elsewhere,
+      VEHICLE_STATES.reserved,
+      VEHICLE_STATES.on_trip,
+      VEHICLE_STATES.non_operational
+    ],
+    [VEHICLE_EVENTS.decommissioned]: [VEHICLE_STATES.removed],
+    [VEHICLE_EVENTS.provider_drop_off]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.unspecified]: [VEHICLE_STATES.available, VEHICLE_STATES.removed]
   }
 }
 
-const getNextState = (currStatus: VEHICLE_STATE, nextEvent: VEHICLE_EVENT): VEHICLE_STATE | undefined => {
+const getNextStates = (currStatus: VEHICLE_STATE, nextEvent: VEHICLE_EVENT): Array<VEHICLE_STATE> | undefined => {
   return stateTransitionDict[currStatus]?.[nextEvent]
 }
 
+function isEventSequenceValidHelper(eventTypeA: VEHICLE_EVENT, eventTypeB: VEHICLE_EVENT) {
+  const currStates = EVENT_STATES_MAP[eventTypeA]
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  for (const currState of currStates) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    // See if it's possible to transition to any states using eventB's event_type
+    const nextStates: any = getNextStates(currState, eventTypeB)
+    if (nextStates) {
+      return true
+    }
+  }
+  return false
+}
+
+function isEventSequenceValid(eventA: VehicleEvent, eventB: VehicleEvent) {
+  for (const eventTypeA of eventA.event_types) {
+    for (const eventTypeB of eventB.event_types) {
+      if (isEventSequenceValidHelper(eventTypeA, eventTypeB)) {
+        return true
+      }
+    }
+  }
+  return false
+}
```

Comment:
Ok, I think the thing you want me to write basically validates state-state transitions, but the test I want to repair does event-event transitions. E.g. instead of verifying that you can go from `available` to `unknown` via `unspecified`, the purpose of the test is to verify that that there's a path bewteen `comms_lost` and `comms_restored`.

The test you want is worth adding, but I need to update `VehicleEvent` to have a `state` field, which I would prefer to do in a different PR.

Reply:
I feel like there's a misunderstanding about what this function is supposed to do. If you look at the one place it's called, it's supposed to validate event transitions, e.g. that there's a path from `comms_lost` to `comms_restored`.

Reply:
Ok, I think we're talking past each other a bit. The purpose of the original test I was trying to repair is to verify that `stateTransitionDict` is accurate by comparing it to `state-transition-expected`, which literally _is_ an event to event mapping. Now, if you think that's a stupid test and we should delete the whole thing, that could be reasonable, but I don't think I really have a solid idea what I'm supposed to replace it with. Now that we no longer have a 1:1 mapping between events and states, the event-event mapping actually seems like an even better test than it did before, since you can have multiple edges going into the same state, and multiple edges coming out.